src/AutoSaveStatus/AutoSaveStatusProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { PropsWithChildren, useCallback, useEffect, useRef, useState } from "react";
+
+export enum AutoSaveStatus {
+  IDLE = "idle",
+  SAVING = "saving",
+  DONE = "done",
+  ERROR = "error",
+}
+
+export interface AutoSaveStatusContextType {
+  status: AutoSaveStatus;
+  /** Resets status to IDLE, particularly useful if "Error" or "Done" is stale */
+  resetStatus: VoidFunction;
```

Comment:
Just in terms as-minimal-as-possible API, do you think we need to expose `resetStatus`? In theory who would call it?

Reply:
Ah yeah, that makes sense. Was thinking that seems more like a private API, vs. something we'd an a caller to call, but np if we don't want to make that distinction.